src/image-copy/azext_imagecopy/custom.py

```diff
@@ -45,19 +44,31 @@ def imagecopy(cmd, source_resource_group_name, source_object_name, target_locati
                                    '--resource-group', source_resource_group_name],
                                   only_show_errors=only_show_errors)
 
+    # get detailed information of target image
     json_cmd_output = run_cli_command(cli_cmd, return_as_json=True)
 
+    # check if the source image has data disks
     if json_cmd_output['storageProfile']['dataDisks']:
         logger.warning(
             "Data disks in the source detected, but are ignored by this extension!")
 
     source_os_disk_id = None
     source_os_disk_type = None
 
+    # get the os disk id from source image
     try:
         source_os_disk_id = json_cmd_output['storageProfile']['osDisk']['managedDisk']['id']
         if source_os_disk_id is None:
             raise TypeError
+        else:
+            try:
+                cli_cmd = prepare_cli_command(['disk', 'show', '--ids', source_os_disk_id],
+                                              output_as_json=False,
+                                              only_show_errors=only_show_errors)
+                run_cli_command(cli_cmd)
+            except:
+                raise CLIError('Unable to find the source OS disk. Please make sure the source OS disk is not deleted.\n'
+                               'If you deleted the source disk where the image was created (or chose to delete the VM while creating the image). Please refer to https://github.com/Azure/azure-cli/issues/25431 for temporary solution.')
```

Comment:
Please use the specific error type `ResourceNotFoundError` from `azure.cli.core.azclierror` instead of `CLIError`
```suggestion
                raise ResourceNotFoundError('Unable to find the source OS disk. Please make sure the source OS disk is not deleted.\n'
                               'If you deleted the source disk where the image was created (or chose to delete the VM while creating the image). Please refer to https://github.com/Azure/azure-cli/issues/25431 for temporary solution.')
```